app/models/user.rb

```diff
@@ -0,0 +1,15 @@
+class User < ApplicationRecord
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+  has_many :posts, foreign_key: 'author_id', dependent: :destroy
+
+  validates :name, presence:
+
+  def recent_posts(limit = 3)
+    posts.order(created_at: :desc).limit(limit)
+  end
+
+  def update_posts_counter
+    update(posts_counter: posts.count)
+  end
```

Comment:
- [ ] Hi @Omwomo, based on the requirements of this project, the User model should have only one method. Method that returns 3 most recent posts :+1:  `update_posts_counter` should have been implemented in the Post model. Let's kindly fix :wrench: this issues.